engine/access/rpc/backend/backend_block_details.go

```diff
@@ -58,3 +58,15 @@ func (b *backendBlockDetails) GetBlockByHeight(_ context.Context, height uint64)
 
 	return block, nil
 }
+
+func (b *backendBlockDetails) GetBlockStatus(_ context.Context, id flow.Identifier) flow.BlockStatus {
+	block, err := b.blocks.ByID(id)
```

Comment:
since you're only checking if the block exists, it's best to use `b.headers.ByID(id)` since it requires fewer db lookups and uses a cache

Reply:
I think it would make even more sense to skip this step completely by passing the block/blockheader into getBlockStatus because the lookups for get block / get header would have already been performed by the time we're checking block status.